src/network/networkd-netdev-wireguard.c

```diff
@@ -0,0 +1,625 @@
+        Wireguard *w;
+
+        assert(data);
+
+        w = WIREGUARD(data);
+
+        assert(w);
+
+        return parse_wireguard_key(unit,
+                                   filename,
+                                   line,
+                                   section,
+                                   section_line,
+                                   lvalue,
+                                   ltype,
+                                   rvalue,
+                                   data,
+                                   &w->dev.preshared_key);
+}
+
+
+int config_parse_wireguard_public_key(const char *unit,
+                                      const char *filename,
+                                      unsigned line,
+                                      const char *section,
+                                      unsigned section_line,
+                                      const char *lvalue,
+                                      int ltype,
+                                      const char *rvalue,
+                                      void *data,
+                                      void *userdata) {
+        Wireguard *w;
+        WireguardPeer *peer;
+
+        assert(data);
+
+        w = WIREGUARD(data);
+
+        assert(w);
+
+        peer = new_peer(w, section_line);
+        if (!peer)
+                return -ENOMEM;
+
+        return parse_wireguard_key(unit,
+                                   filename,
+                                   line,
+                                   section,
+                                   section_line,
+                                   lvalue,
+                                   ltype,
+                                   rvalue,
+                                   data,
+                                   peer->fields.public_key);
+}
+
+int config_parse_wireguard_allowed_ips(const char *unit,
+                                       const char *filename,
+                                       unsigned line,
+                                       const char *section,
+                                       unsigned section_line,
+                                       const char *lvalue,
+                                       int ltype,
+                                       const char *rvalue,
+                                       void *data,
+                                       void *userdata) {
+        _cleanup_strv_free_ char **words = NULL;
+        char **word;
+        const char *address, *end;
+        union in_addr_union buffer;
+        unsigned char prefixlen;
+        int r, family;
+        Wireguard *w;
+        WireguardPeer *peer;
+        WireguardIpmask *ipmask;
+
+        assert(rvalue);
+        assert(data);
+
+        w = WIREGUARD(data);
+
+        peer = new_peer(w, section_line);
+        if (!peer)
+                return -ENOMEM;
+
+        r = strv_split_extract(&words, rvalue, ",", 0);
```

Comment:
so, you are not actually interested in the "words" array, right? you just want to split this up, parse the individual words and use the parsed values, but throw the words away, right?

First of all, there's a memory leak here, you never free "words"... That said, I#d suggest you do away entirely with this, and instead change this to be a loop around extract_first_word(). i.e. instead of first placing this in an strv, and then iterating through that, please just iterate each word directly, without placing it in any strv...